include/stl2/view/iterator_range.hpp

```diff
@@ -0,0 +1,227 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Eric Niebler 2017
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_VIEW_ITERATOR_RANGE_HPP
+#define STL2_VIEW_ITERATOR_RANGE_HPP
+
+#include <stl2/detail/fwd.hpp>
+#include <stl2/detail/iterator/concepts.hpp>
+#include <stl2/detail/range/access.hpp>
+#include <stl2/detail/range/concepts.hpp>
+#include <stl2/view/view_interface.hpp>
+#include <stl2/detail/algorithm/tagspec.hpp>
+#include <stl2/utility.hpp>
+
+#include <tuple>
+
+STL2_OPEN_NAMESPACE {
+	namespace ext {
+		template <Iterator I, Sentinel<I> S = I>
+		struct iterator_range
+			: tagged_pair<tag::begin(I), tag::end(S)>,
+			  view_interface<iterator_range<I, S>> {
+		public:
+			using iterator = I;
+			using sentinel = S;
+			iterator_range() = default;
+
+			constexpr iterator_range(I i, S s)
+			: tagged_pair<tag::begin(I), tag::end(S)>{i, s} {}
+
+			template <ConvertibleTo<I> X, ConvertibleTo<S> Y>
+			constexpr iterator_range(iterator_range<X, Y> r)
```

Comment:
One user-defined conversion instead of 2.

Reply:
Ah, I missed that the conversion from `iterator_range<X, Y>` to `pair<X, Y>` triggers the user-defined conversion operator instead of using the standard conversion-to-base. I could suggest disabling that operator for the same-type case, but that's more a tradeoff than a clear win. Nevermind.